File: jscomp/test/bs_map_set_dict_test.js
'use strict';

var Mt = require("./mt.js");
var Belt_Id = require("../../lib/js/belt_Id.js");
var Belt_Map = require("../../lib/js/belt_Map.js");
var Belt_List = require("../../lib/js/belt_List.js");
var Belt_Array = require("../../lib/js/belt_Array.js");
var Belt_MapInt = require("../../lib/js/belt_MapInt.js");
var Belt_MapDict = require("../../lib/js/belt_MapDict.js");
var Belt_SetDict = require("../../lib/js/belt_SetDict.js");
var Caml_primitive = require("../../lib/js/caml_primitive.js");
var Array_data_util = require("./array_data_util.js");

var suites = {
  contents: /* [] */0
};

var test_id = {
  contents: 0
};

function eq(loc, x, y) {
  return Mt.eq_suites(test_id, suites, loc, x, y);
}

function b(loc, v) {
  return Mt.bool_suites(test_id, suites, loc, v);
}

var Icmp = Belt_Id.comparable(Caml_primitive.caml_int_compare);

var Icmp2 = Belt_Id.comparable(Caml_primitive.caml_int_compare);

var Ic3 = Belt_Id.comparable(Caml_primitive.caml_int_compare);

var m0_cmp = Icmp.cmp;

var m0 = {
  cmp: m0_cmp,
  data: Belt_MapDict.empty
};

var m00_cmp = Ic3.cmp;

var m00 = {
  cmp: m00_cmp,
  data: undefined
};

var I2 = Belt_Id.comparable((function (x, y) {
        return Caml_primitive.caml_int_compare(y, x);
      }));

var m_cmp = Icmp2.cmp;

var m = {
  cmp: m_cmp,
  data: Belt_MapDict.empty
};

var m2_cmp = I2.cmp;

var m2 = {
  cmp: m2_cmp,
  data: Belt_MapDict.empty
};

var data = Belt_MapDict.empty;

Belt_Map.getId(m2);

var m_dict = Belt_Map.getId(m);

for(var i = 0; i <= 100000; ++i){
  data = Belt_MapDict.set(data, i, i, m_dict.cmp);
}

var data$1 = data;

var newm_cmp = m_dict.cmp;

var newm = {
  cmp: newm_cmp,
  data: data$1
};

console.log(newm);

var m11 = Belt_MapDict.set(Belt_MapDict.empty, 1, 1, Icmp.cmp);

console.log(m11);

var m_dict$1 = Belt_Map.getId(m);

var cmp = m_dict$1.cmp;

var data$2;

for(var i$1 = 0; i$1 <= 100000; ++i$1){
  data$2 = Belt_SetDict.add(data$2, i$1, cmp);
}

console.log(data$2);

function f(param) {
  return Belt_Map.fromArray(param, Icmp);
}

function $eq$tilde(a, b) {
  return (function (param) {
      return Belt_Map.eq(a, b, param);
    });
}

var u0 = f(Belt_Array.map(Array_data_util.randomRange(0, 39), (function (x) {
            return /* tuple */[
                    x,
                    x
                  ];
          })));

var u1 = Belt_Map.set(u0, 39, 120);

b("File \"bs_map_set_dict_test.ml\", line 80, characters 4-11", Belt_Array.every2(Belt_MapDict.toArray(u0.data), Belt_Array.map(Array_data_util.range(0, 39), (function (x) {
                return /* tuple */[
                        x,
                        x
                      ];
              })), (function (param, param$1) {
            if (param[0] === param$1[0]) {
              return param[1] === param$1[1];
            } else {
              return false;
            }
          })));

b("File \"bs_map_set_dict_test.ml\", line 85, characters 4-11", Belt_List.every2(Belt_MapDict.toList(u0.data), Belt_List.fromArray(Belt_Array.map(Array_data_util.range(0, 39), (function (x) {
                    return /* tuple */[
                            x,
                            x
                          ];
                  }))), (function (param, param$1) {
            if (param[0] === param$1[0]) {
              return param[1] === param$1[1];
            } else {
              return false;
            }
          })));

eq("File \"bs_map_set_dict_test.ml\", line 90, characters 5-12", Belt_Map.get(u0, 39), 39);

eq("File \"bs_map_set_dict_test.ml\", line 91, characters 5-12", Belt_Map.get(u1, 39), 120);

var u = f(Belt_Array.makeByAndShuffle(10000, (function (x) {
            return /* tuple */[
                    x,
                    x
                  ];
          })));

eq("File \"bs_map_set_dict_test.ml\", line 97, characters 4-11", Belt_Array.makeBy(10000, (function (x) {
            return /* tuple */[
                    x,
                    x
                  ];
          })), Belt_MapDict.toArray(u.data));

Mt.from_pair_suites("Bs_map_set_dict_test", suites.contents);

var M = /* alias */0;

var MI = /* alias */0;

var I = /* alias */0;

var A = /* alias */0;

var L = /* alias */0;

var vv = Belt_MapInt.empty;

var vv2 = Belt_MapInt.empty;

var Md0 = /* alias */0;

var ISet = /* alias */0;

var S0 = /* alias */0;

exports.suites = suites;
exports.test_id = test_id;
exports.eq = eq;
exports.b = b;
exports.Icmp = Icmp;
exports.Icmp2 = Icmp2;
exports.Ic3 = Ic3;
exports.M = M;
exports.MI = MI;
exports.I = I;
exports.A = A;
exports.L = L;
exports.m0 = m0;
exports.m00 = m00;
exports.I2 = I2;
exports.m = m;
exports.m2 = m2;
exports.vv = vv;
exports.vv2 = vv2;
exports.Md0 = Md0;
exports.ISet = ISet;
exports.S0 = S0;
exports.f = f;
exports.$eq$tilde = $eq$tilde;
/* Icmp Not a pure module */

File: jscomp/test/bs_map_set_dict_test.ml
let suites :  Mt.pair_suites ref  = ref []
let test_id = ref 0
let eq loc x y = Mt.eq_suites ~suites ~test_id loc x y 
let b loc v  = Mt.bool_suites ~suites ~test_id loc v 

module Icmp = 
  (val Belt.Id.comparable
    ~cmp:(fun (x : int) y -> 
      compare x y
     )
  )
module Icmp2 = 
(val Belt.Id.comparable ~cmp:(fun  (x : int) y ->
      compare x y ))
  
module M = Belt.Map
module MI = Belt.Map.Int
(* module B = Belt.Bag *)
module I = Array_data_util
module A = Bs_Array
module L = Belt.List
let m0 : (_,string,_) M.t = M.make (module Icmp)   

  
module I2 = 
(val Belt.Id.comparable ~cmp:(fun  (x : int) y -> compare y x ))

  
let m = M.make (module Icmp2)
let m2 : (int, string, _) M.t = M.make (module I2)
let vv = MI.empty 
let vv2 = MI.empty
module Md0 = Belt.Map.Dict
let () = 
  let count = 1_000_00 in 
  let data = ref (M.getData m) in 
  let m2_dict, m_dict = M.(getId m2, getId m) in 
  let module N = (val m2_dict) in 
  let module Mm = ( val m_dict) in
  for i = 0 to count do 
    data := 
      Md0.set !data 
      ~cmp:  Mm.cmp
      i i 
  done ;
  let newm = M.packIdData ~data:!data ~id:m_dict in 
  Js.log newm
module ISet = Belt.Set 
let () =     
  let  m = Md0.empty in 
  let m11 = 
    Md0.set ~cmp:Icmp.cmp m
    1 1 
  in  
  let _m20 = M.make (module Icmp) in 
  Js.log m11

module S0 = Belt.Set.Dict  
let () =   
 let count = 100_000 in 
  let v = ISet.make (module Icmp2) in 
  let m_dict = M.getId m in 
  let module M = (val m_dict) in 
  let cmp = M.cmp in 
  let data = ref (ISet.getData v) in 
  for i = 0 to count do 
    data := S0.add ~cmp !data i 
  done ;
  Js.log !data  

let f = M.ofArray ~id:(module Icmp)
let (=~) a b = M.eq a b  

let () =   
  let u0 =  f (A.map (I.randomRange 0 39) (fun x -> (x,x))) in  
  let u1 = M.set u0 39 120 in 
  b __LOC__
  (A.every2 (M.toArray u0) 
   (A.map (I.range 0 39) (fun x -> (x,x)))
   (fun (x0,x1) (y0,y1) -> x0 = y0 && x1 = y1));
  
  b __LOC__
  (L.every2 
    (M.toList u0)
    (L.ofArray (A.map (I.range 0 39) (fun  x -> (x,x))))
    (fun (x0,x1) (y0,y1) -> x0 = y0 && x1 = y1));
  eq __LOC__ (M.get u0 39) (Some 39);
  eq __LOC__ (M.get u1 39) (Some 120)


let () =     
  let u = f 
    ((A.makeByAndShuffle 10_000 (fun x  -> (x,x)))) in 
 eq __LOC__    
  (A.makeBy 10_000 (fun x  -> (x,x)))
  (M.toArray u)


;; Mt.from_pair_suites __FILE__ !suites  
